group-coordinator/src/main/java/org/apache/kafka/coordinator/group/assignor/GroupSpec.java

```
@@ -39,4 +41,20 @@ public interface GroupSpec {
      *         False, otherwise.
      */
     boolean isPartitionAssigned(Uuid topicId, int partitionId);
+
+    /**
+     * Gets the member subscription specification for a member.
+     *
+     * @param memberId          The member Id.
+     * @return The member's subscription metadata.
+     */
+    MemberSubscriptionSpec memberSubscriptionSpec(String memberId);
```

nit: Let's call it `memberSubscription`. We should also precise in the javadoc what we do if the member id does not exist. I suppose that we return null.

---

thanks for the comment! I wanted to say that during our design discussions, we had agreed to keep it as memberSubscriptionSpec to maintain consistency and streamline the review process so we don't go back and forth. If there are new considerations or changes that we might not have anticipated, I would love to understand them better so we can make this more efficient going forward.

---

I actually wanted to asak about whether we want to return null or return an empty object. I returned a new memberSubscriptionSpecImpl object in the impl

---

discussed offline, we want to throw an IllegalStateException when the memberId is not found